### PR TITLE
Continues processing JSONs even if hook fails (fixes #1038)

### LINKED ISSUE #1038: `pretty-format-json` stops formatting JSONs when it encounters one with incorrect format (e.g. JSON5 comments)
Transferred from https://github.com/pre-commit/pre-commit/issues/3124.

`pretty-format-json` prints "Input File {json_file} is not a valid JSON, consider using check-json" when it encounters an incorrectly formatted JSON file (e.g. with JSON5 comments), but it's not clear from that message that it then stops further processing files.

As `pre-commit` can partition the processed files into batches, this makes it even harder to tell as multiple batches of files can fail midway through with a few "X is not a valid JSON" messages printed, giving the impression that all the JSONs were formatted (or attempted formatted) and only those files printed are incorrect. The reality is that some JSONs may not have been formatted. I think this hook should try to format every JSON and only fail at the end if any have an invalid JSON format.

### PR DESCRIPTION
**Pull request by amarvin**
Fixes #1038

### CODE REVIEW

**Comment by amarvin** [comment]
@asottile, any chance you could consider this change, please?

File: pre_commit_hooks/pretty_format_json.py [hunk 1]
```diff
@@ -115,7 +115,8 @@ def main(argv: Sequence[str] | None = None) -> int:
                 f'Input File {json_file} is not a valid JSON, consider using '
                 f'check-json',
             )
-            return 1
+            status = 1
+            pretty_contents = contents
```
Comment (asottile) [inline pre_commit_hooks/pretty_format_json.py:119]: this should use `try` `else`

**Review by asottile — COMMENTED** [review]
this needs a test which is why I didn't look at it

**Comment by amarvin** [comment]
> this needs a test which is why I didn't look at it

Great point! I added a test that failed in `main`, but passes in this branch.

File: tests/pretty_format_json_test.py [hunk 2]
```diff
@@ -82,6 +82,28 @@ def test_autofix_main(tmpdir):
     assert ret == 0
 
 
+def test_invalid_main(tmpdir):
```
Comment (amarvin) [inline tests/pretty_format_json_test.py:85]: Could also consider combining this test with the one above.

**Review by asottile — APPROVED** [review]

**Pull request merged**